src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyDiscoveryManager.cs

```diff
@@ -241,12 +235,24 @@ public void Close()
     /// <inheritdoc/>
     public void HandleDiscoveryComplete(DiscoveryCompleteEventArgs discoveryCompleteEventArgs, IEnumerable<TestCase> lastChunk)
     {
-        _baseTestDiscoveryEventsHandler.HandleDiscoveryComplete(discoveryCompleteEventArgs, lastChunk);
+        // When discovery is complete then the last discovered source is still marked
+        // as partially discovered, so we need to mark it as fully discovered.
+        _discoverySourceStatusCache.MarkTheLastChunkSourcesAsFullyDiscovered(lastChunk);
```

Comment:
From the various tests I made with the playground project, I haven't found any case where this would be useful. I don't know if that's ok to remove or if there is a particular case I missed.